### PR TITLE
implement #18

### PR DESCRIPTION
**Pull request by surf1nb1rd**
Hi,
I am submitting a proposal for the task.

### CODE REVIEW

File: chronicles.nim [hunk 1]
```diff
@@ -228,10 +238,13 @@ macro logIMPL(Stream: typed,
     # in `log_output` to kick in.
     let recordRef = if recordArity == 1: record
                     else: newTree(nnkBracketExpr, record, newLit(i))
+    var filename = lineInfo.filename & ":" & $lineInfo.line
     code.add quote do:
       initLogRecord(`recordRef`, LogLevel(`severity`),
                     `topicsNode`, `eventName`)
       setFirstProperty(`recordRef`, "thread", `threadId`)
+      if `lineNumbersNN`:
```
Comment (zah) [inline chronicles.nim]: This doesn't need to be in the generated code. Put the `if` outside of the `quote` block:

``` nim
if useLineNumbers:
  code.add newCall("setProperty", recordRef, ...)
```

File: chronicles.nim [hunk 2]
```diff
@@ -240,14 +253,14 @@ macro logIMPL(Stream: typed,
   code.add newCall("flushRecord", record)
 
   result = newBlockStmt(id"chroniclesLogStmt", code)
-
+  echo result.repr
```
Comment (zah) [inline chronicles.nim]: oops, debugging code left in the commit.

File: chronicles.nim [hunk 3]
```diff
@@ -197,7 +201,12 @@ macro logIMPL(Stream: typed,
         enabledTopicsMatch = true
       elif t in requiredTopics:
         dec requiredTopicsCount
-
+  
+  # Handling file name and line numbers on/off (lineNumbersEnabled) for particular log statements
+  if finalBindings.hasKey("chroniclesLineNumbers"):
+    lineNumbersSpec = $finalBindings["chroniclesLineNumbers"] == "true"
```
Comment (kdeme) [inline chronicles.nim]: This approach results in  "true" / "false" and true/false as allowed values. Perhaps this is intended, but originally true/false was requested in the issue. Allowing also "true" / "false" could give issues in the future if this is not stated in the documentation for example. So my suggestion would be to either change this or change the spec (and future documentation).

There is also no checking done if the value is anything else than these values, it will allow any other string and return it as false. I would rather like to see an error if I typed "truee".

File: chronicles.nim [hunk 4]
```diff
@@ -197,7 +201,12 @@ macro logIMPL(Stream: typed,
         enabledTopicsMatch = true
       elif t in requiredTopics:
         dec requiredTopicsCount
-
+
```
Comment (kdeme) [inline chronicles.nim]: nitpick: remove unnecessary whitespace

File: chronicles.nim [hunk 5]
```diff
@@ -197,7 +201,12 @@ macro logIMPL(Stream: typed,
         enabledTopicsMatch = true
       elif t in requiredTopics:
         dec requiredTopicsCount
-
+  
+  # Handling file name and line numbers on/off (lineNumbersEnabled) for particular log statements
+  if finalBindings.hasKey("chroniclesLineNumbers"):
+    lineNumbersSpec = $finalBindings["chroniclesLineNumbers"] == "true"
+    finalBindings.del("chroniclesLineNumbers")
+
```
Comment (kdeme) [inline chronicles.nim]: nitpick: remove unnecessary whitespace

**Comment by surf1nb1rd** [comment]
Thanks for your comments, @kdeme and @zah, I addressed most of them and added a description of the option in the README. I plan to work on some of the other nim-chronicles bounties, as well, and will make sure to post when I start, so that we don't end up working on the same thing. I would like to do #19 next, unless you have already started working on it, @kdeme?

I spent about 6 hours on the task with getting to know the library, fixing the changes etc. 
ETH would be fine, thanks, my address is:
0xc4Ca552F7A2aB84bfb4F1dc3773Bb5b09E6Def5A

**Comment by kdeme** [comment]
@surf1nb1rd, I have not looked into #19 yet so go ahead. Perhaps mention it in the actual issue also so that it is clear for others too.

**Comment by zah** [comment]
Thanks for the fixes @surf1nb1rd. For #19, the bounty might need to get some upvotes from the team first (each bounty needs 3 upvotes before it's considered accepted).

**Comment by zah** [comment]
@kdeme, may I suggest https://github.com/status-im/nim-chronicles/issues/17 as another relatively simple bounty that touches pretty much the same areas in the code as #18. If you deliver it, it would be only fair to reward some of the time you put to make the contributions here.

**Comment by pablanopete** [comment]
Confirmed with @zah work done and time spent, payment sent - https://etherscan.io/tx/0xa2b1ace30b5dc56d8bb6d06629d26aea9392cbf6d0369ebe68f98c90f4ceae77

**Comment by kdeme** [comment]
@zah, OK, I will go for #17. Thanks.
